echopype/commongrid/api.py

```diff
@@ -214,5 +221,108 @@ def compute_MVBS_index_binning(ds_Sv, range_sample_num=100, ping_num=100):
+    mean Sv and the mean height within each cell.
+
+    In addition, here the binning of pings into individual cells is based on the actual horizontal
+    distance computed from the latitude and longitude coordinates of each ping in the Sv dataset.
+    Therefore, both regular and irregular horizontal distance in the Sv dataset are allowed.
+    This is different from Echoview's assumption of constant ping rate, vessel speed, and sample
+    thickness when computing mean Sv.
+    """
+    # Check Sv contains lat/lon
+    if "latitude" not in ds_Sv or "longitude" not in ds_Sv:
+        raise ValueError("Both 'latitude' and 'longitude' must exist in the input Sv dataset.")
+
+    # Check if depth vectors are identical within each channel
+    if not ds_Sv["depth"].groupby("channel").map(check_identical_depth).all():
+        raise ValueError(
+            "Only Sv data with identical depth vectors across all pings "
+            "are allowed in the current compute_NASC implementation."
+        )
+
+    # Get distance from lat/lon in nautical miles
+    dist_nmi = get_distance_from_latlon(ds_Sv)
+
+    # Find binning indices along distance
+    bin_num_dist, dist_bin_idx = get_dist_bin_info(dist_nmi, cell_dist)  # dist_bin_idx is 1-based
+
+    # Find binning indices along depth: channel-dependent
+    bin_num_depth, depth_bin_idx = get_depth_bin_info(ds_Sv, cell_depth)  # depth_bin_idx is 1-based
+
+    # Compute mean sv (volume backscattering coefficient, linear scale)
+    # This is essentially to compute MVBS over a the cell defined here,
+    # which are typically larger than those used for MVBS.
+    # The implementation below is brute force looping, but can be optimized
+    # by experimenting with different delayed schemes.
+    # The optimized routines can then be used here and
+    # in commongrid.compute_MVBS and clean.estimate_noise
+    sv_mean = []
+    for ch_seq in np.arange(ds_Sv["channel"].size):
+        # TODO: .compute each channel sequentially?
+        #       dask.delay within each channel?
+        ds_Sv_ch = ds_Sv["Sv"].isel(channel=ch_seq).data  # preserve the underlying type
+
+        sv_mean_dist_depth = []
+        for dist_idx in np.arange(bin_num_dist) + 1:  # along ping_time
+            sv_mean_depth = []
+            for depth_idx in np.arange(bin_num_depth) + 1:  # along depth
+                # Sv dim: ping_time x depth
+                Sv_cut = ds_Sv_ch[dist_idx == dist_bin_idx, :][
+                    :, depth_idx == depth_bin_idx[ch_seq]
+                ]
+                sv_mean_depth.append(np.nanmean(10 ** (Sv_cut / 10)))
+            sv_mean_dist_depth.append(sv_mean_depth)
+
+        sv_mean.append(sv_mean_dist_depth)
+
+    # Compute mean height
+    # For data with uniform depth step size, mean height = vertical size of cell
+    height_mean = cell_depth
+    # TODO: generalize to variable depth step size
+
+    ds_NASC = xr.DataArray(
```

Comment:
`ds_NASC` is a DataArray. But the variable name and the function signature (`-> xr.Dataset`) and docstring point to an expectation of an xr.Dataset.

Reply:
Oh right, I forgot about this. I was initially thinking we might need to add other info (in the form of data variables) for NASC. 

Looking at what we have for `compute_MVBS`, I think we probably need to add `frequency_nominal` (coordinate `channel`). Seems something we should keep around, and there's not actual difference once it's saved since things are always saved as a dataset. What do you think?

Reply:
Definitely, let's add `frequency_nominal`!

Reply:
Ok, I will add that now along with the docstring addition of requiring `depth`.

Reply:
Great. I see that this is taken care of. I'll pull this into my local clone and work on the attributes.

Reply:
The conversion to Dataset fails b/c the DataArray requires a name. I've fixed this locally so I can work on adding the attributes.

Also, the test needs to be updated now that a dataset is returned. I've fixed that locally, too.